src/main/java/de/hysky/skyblocker/utils/NEURepoManager.java

```diff
@@ -58,28 +64,36 @@ private static CompletableFuture<Void> loadRepository() {
                     LOGGER.info("[Skyblocker] NEU Repository Downloaded");
                 }
                 NEU_REPO.reload();
-            } catch (TransportException e){
+                return true;
+            } catch (TransportException e) {
                 LOGGER.error("[Skyblocker] Transport operation failed. Most likely unable to connect to the remote NEU repo on github", e);
             } catch (RepositoryNotFoundException e) {
                 LOGGER.warn("[Skyblocker] Local NEU Repository not found or corrupted, downloading new one", e);
-                deleteAndDownloadRepository();
+                Scheduler.INSTANCE.schedule(() -> deleteAndDownloadRepository(MinecraftClient.getInstance().player), 1);
             } catch (Exception e) {
                 LOGGER.error("[Skyblocker] Encountered unknown exception while initializing NEU Repository", e);
             }
+            return false;
         });
     }
 
-    private static void deleteAndDownloadRepository() {
-        CompletableFuture.runAsync(() -> {
+    private static void deleteAndDownloadRepository(PlayerEntity player) {
+        if (REPO_LOADING != null && !REPO_LOADING.isDone()) {
+            player.sendMessage(Constants.PREFIX.get().append(Text.translatable("skyblocker.updateRepository.loading")), false);
+            return;
+        }
+        player.sendMessage(Constants.PREFIX.get().append(Text.translatable("skyblocker.updateRepository.start")), false);
+
+        REPO_LOADING = CompletableFuture.runAsync(() -> {
             try {
                 ItemRepository.setFilesImported(false);
                 FileUtils.recursiveDelete(NEURepoManager.LOCAL_REPO_DIR);
-            } catch (Exception ex) {
-                if (MinecraftClient.getInstance().player != null)
-                    MinecraftClient.getInstance().player.sendMessage(Constants.PREFIX.get().append(Text.translatable("skyblocker.updaterepository.failed")), false);
-                return;
+                player.sendMessage(Constants.PREFIX.get().append(Text.translatable("skyblocker.updateRepository.deleted")), false);
+                player.sendMessage(Constants.PREFIX.get().append(Text.translatable(loadRepository().join() ? "skyblocker.updateRepository.success" : "skyblocker.updateRepository.failed")), false);
```

Comment:
I assume this wouldn't block the render thread?

Reply:
Yeah, it's in a completable future.